src/helpers/localStorage.ts

```diff
@@ -0,0 +1,30 @@
+export interface LocalStorageKeys {
+  AUTORUN_SIMULATION: string
+  LANG: string
+  LOG_SCALE: string
+  SHOW_HUMANIZED_RESULTS: string
+  SUPPRESS_DISCLAIMER: string
+}
+
+export const LOCAL_STORAGE_KEYS: LocalStorageKeys = {
+  AUTORUN_SIMULATION: 'autorun-simulation',
+  LANG: 'lang',
+  LOG_SCALE: 'log-scale',
+  SHOW_HUMANIZED_RESULTS: 'show-humanized-results',
+  SUPPRESS_DISCLAIMER: 'suppress-disclaimer',
+}
+
+function set<T>(key: string, value: T): void {
+  localStorage.setItem(key, typeof value !== 'string' ? JSON.stringify(value) : value)
+}
+
+function get<T>(key: string): T | null {
+  const value = localStorage.getItem(key)
+  try {
```

Comment:
I'm not aware of any way in Typescript to assert the typeof T at runtime to prevent attempting to JSON.parse something unparseable, like a normal string, for example the language.

If someone has a suggestion as to how the type safety here can be improved, would love to learn how!

Reply:
Yep, TS does not have any runtime, so runtime validation should be performed.
We validate form params declaratively with `yup`:
https://github.com/neherlab/covid19_scenarios/blob/master/src/components/Main/validation/schema.ts

So this part might be shared. And I thing URL serialization/deserialization folks in https://github.com/neherlab/covid19_scenarios/issues/31 may add `io-ts`. Which is very typescript-y and awesome.

But something imperative like a good old `validate.js` or in simple cases even `lodash` should do.